pink/tasks/relative_frame_task.py

```diff
@@ -86,15 +86,19 @@ def set_position_cost(
         """
         if isinstance(position_cost, float):
             assert position_cost >= 0.0
-        else:  # not isinstance(position_cost, float)
-            assert all(cost >= 0.0 for cost in position_cost)
-        if isinstance(self.cost, np.ndarray):
-            self.cost[0:3] = position_cost
-        else:  # self.cost is not a vector
-            raise TaskDefinitionError(
-                "Frame task cost should be a vector, "
-                f"currently cost={self.cost}"
-            )
+        else:  # Should be a ndarray or seq
+            if not isinstance(position_cost, np.ndarray):  # Must be seq
+                try:
+                    position_cost = np.array(position_cost)
+                except Exception:  # Not a proper float sequence
+                    raise TaskDefinitionError(
+                        "Position task cost should be a float or a "
+                        "seq of float or ndarray of size 1 or 3,"
+                        f"currently cost={self.cost}"
+                    )
+                assert np.all(position_cost >= 0.0)
```

Comment:
Oh I see, you do mean `else: if`, and then the level of indentation is wrong.

Alternative suggestion: write `self.cost[0:3]` first, then do the check in the end.

Reply:
I think it is better to set at the end, to avoid setting negative value if the user use a try/except.